test/core/crawlers/browser_crawler.test.ts

```diff
@@ -666,6 +666,35 @@ describe('BrowserCrawler', () => {
         expect(called).toBeTruthy();
     });
 
+    test('should increment session usage correctly', async () => {
+        const sessionUsageHistory: number[] = [];
+
+        const browserCrawler = new BrowserCrawlerTest({
+            browserPoolOptions: {
+                browserPlugins: [puppeteerPlugin],
+            },
+            useSessionPool: true,
+            sessionPoolOptions: {
+                maxPoolSize: 1,
+            },
+            requestHandler: async ({ session }) => {
+                // @ts-expect-error Accessing private property
```

Comment:
Since `usageCount` is private, should we even test for this? This whole is rather a code quality nit to me than a huge fix with user impact.

Reply:
Testing private stuff does not make much sense. However, maybe we could add a read only getter for that property - it might be helpful for introspection and what not

Reply:
The latest commit adds public getters for the `Session` fields that appear in the `Session.getState()` anyway. 

While I like that we got rid of many `ts-ignore-error`s in the tests (indicative of the fact that we want to access these fields), I'm not completely sold on the code quality (e.g. underscoring some private fields - those with the getters - while keeping the rest as is). To a naked eye, it looks quite arbitrary and IMO bit messy.

Do you think this is acceptable way to achieve this, or you have a better pattern for those in mind?